fs/fs_zenfs.cc

```diff
@@ -867,17 +947,20 @@ Status ZenFS::DecodeFileUpdateFrom(Slice* slice, bool replace) {
   for (auto it = files_.begin(); it != files_.end(); it++) {
     std::shared_ptr<ZoneFile> zFile = it->second;
     if (id == zFile->GetID()) {
-      std::string oldName = zFile->GetFilename();
+      for (uint32_t i = 0; i < zFile->GetNrLinks(); i++) {
+        if (files_.find(zFile->GetLinkFilename(i)) != files_.end())
+          files_.erase(zFile->GetLinkFilename(i));
+        else
+          return Status::Corruption("DecodeFileUpdateFrom: missing link file");
+      }
 
       s = zFile->MergeUpdate(update, replace);
       update.reset();
 
       if (!s.ok()) return s;
 
-      if (zFile->GetFilename() != oldName) {
-        files_.erase(oldName);
-        files_.insert(std::make_pair(zFile->GetFilename(), zFile));
-      }
+      for (uint32_t i = 0; i < zFile->GetNrLinks(); i++)
+        files_.insert(std::make_pair(zFile->GetLinkFilename(i), zFile));
```

Comment:
Just a thought, what happens if we insert the same name twice? Could happen with corrupt data or error when encoding.

Reply:
We should add a check for that

Reply:
insert would not overwrite or insert a new one, if the same key already exists in the map.

Reply:
You are right. I was certain that `files_` was a vector of pairs. Never mind then.